backend/src/main/java/ca/bc/gov/hlth/mohums/webclient/WebClientService.java

```diff
@@ -46,7 +56,17 @@ public ResponseEntity<List<Object>> getUsersInRole(String clientId, String roleN
 
     // Groups
     public ResponseEntity<Object> getGroups() {
-        return get(groupsPath, null);
+        ResponseEntity<Object> response = get(groupsPath, null);
+        ArrayList<LinkedHashMap> groups = (ArrayList<LinkedHashMap>) response.getBody();
+        assert groups != null;
+        List<Group> groupList = groups.stream().map(g -> getGroupById(g.get("id").toString())).collect(Collectors.toList());
+        return ResponseEntity.of(Optional.of(groupList));
+    }
+    // Group details
+    private Group getGroupById(String groupId) {
+        String path = String.format("%s/%s", groupsPath, groupId);
+        ResponseEntity<Object> response =  get(path, null);
+        return groupMapper.mapToGroup((LinkedHashMap)response.getBody());
```

Comment:
`groupMapper` is used only here in this method, `getGroupById`. `GroupMapper` has no external dependencies, and it is not an expensive object to create. We don't need to mock it to test `WebClientService`.

For these reasons, I recommend _not_ using injection to add `groupMapper` to `WebClientService`, and I don't even recommend making it a class field at all. Instead, either instantiate `groupMapper` right here in the method, or even turn `mapToGroup` into a `static` method so the object doesn't need to be instantiated at all.